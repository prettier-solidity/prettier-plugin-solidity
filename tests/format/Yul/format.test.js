runFormatTest(import.meta, ['slang-yul']);
