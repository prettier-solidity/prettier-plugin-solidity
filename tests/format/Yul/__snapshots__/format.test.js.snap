// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`example1.yul format 1`] = `
====================================options=====================================
parsers: ["slang-yul"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
{
    function power(base, exponent) -> result
    {
        switch exponent
        case 0 { result := 1 }
        case 1 { result := base }
        default
        {
            result := power(mul(base, base), div(exponent, 2))
            switch mod(exponent, 2)
                case 1 { result := mul(base, result) }
        }
    }
}
=====================================output=====================================
{
    function power(base, exponent) -> result {
        switch exponent
        case 0 {
            result := 1
        }
        case 1 {
            result := base
        }
        default {
            result := power(mul(base, base), div(exponent, 2))
            switch mod(exponent, 2)
            case 1 {
                result := mul(base, result)
            }
        }
    }
}
================================================================================
`;

exports[`example2.yul format 1`] = `
====================================options=====================================
parsers: ["slang-yul"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
{
      function power(base, exponent) -> result
     {
        result := 1 for { let i := 0 } lt(i, exponent) { i := add(i, 1) }
        {
            result    :=   mul(result,   base)
        }
    }
}
=====================================output=====================================
{
    function power(base, exponent) -> result {
        result := 1
        for {
            let i := 0
        } lt(i, exponent) {
            i := add(i, 1)
        } {
            result := mul(result, base)
        }
    }
}
================================================================================
`;
